crates/billing-integrations/src/stripe.rs

```diff
@@ -41,32 +41,23 @@ pub struct PublishInvoice {
     fail_fast: bool,
 }
 
-fn parse_date(arg: &str) -> Result<Date, time::error::Parse> {
-    Date::parse(arg, &format_description!("[year]-[month]-[day]"))
+fn parse_date(arg: &str) -> Result<NaiveDate, ParseError> {
+    NaiveDate::parse_from_str(arg, "%Y-%m-%d")
 }
 
-#[derive(Serialize, Deserialize, Debug)]
-struct LineItem {
-    count: f64,
-    subtotal: i64,
-    description: Option<String>,
+#[derive(
+    Debug, Clone, PartialEq, Eq, Hash, sqlx::Type, Deserialize_enum_str, Serialize_enum_str,
+)]
+#[sqlx(rename_all = "snake_case")]
```

Comment:
WRT `serde_enum_str`: the string values for `invoice_type` live in quite a few different places:
* They come from the `invoices_ext` view
* They need to be parsed by sqlx into the right `InvoiceType` enum variant
* Sqlx uses serde, so Serde also needs to know the correct strings to parse/serialize for each variant
* They get set in the Stripe invoice metadata to allow looking up existing invoices instead of always creating new ones
* The front-end also uses them to do some different logic depending on which type of invoice it's rendering

I originally had a manually implemented `to_string` impl, but it felt smelly to have yet another place to manually keep these strings in sync. [Down below I use](https://github.com/estuary/flow/pull/1191/files#diff-4fe0b4ecd5665aaa642a6ced5a704e070710504713aba04f1502804c8aecab0fR174) the `Serialize_enum_str` `to_string()` impl to get me the correct string representation of the invoice type. 

For the same reason, I would like to keep the explicit renames. Reason being, when someone inevitably wants to go look and find all references to those enums, they'll be able to find them easily throughout all the repos, instead of having them be implicit by way of `rename_all="snake_case"`. I feel less strongly about this though, so I'm fine with doing the rename_all if you feel strongly that it should be that way

Reply:
> Sqlx uses serde, so Serde also needs to know the correct strings to parse/serialize for each variant

This is why I made the comment and the source of my confusion: as far as I am aware, if you add a plain Serialize to an enum with empty variants (not structs or tuples), the serde representation _is_ the simple string. For example, here's [CompressionCodec](https://github.com/estuary/flow/blob/master/crates/models/src/journals.rs#L220-L226), which serde's as snake-case variant names ([example here](https://github.com/estuary/flow/blob/master/crates/sources/src/scenarios/snapshots/sources__scenarios__test__collections.snap#L73)).

I'm pretty sure you could just use `serde_json::to_value(my_enum).as_str().unwrap()` if you want the string itself.

> For the same reason, I would like to keep the explicit renames. 

:+1:

Reply:
I'm going to test this right now, but I assumed that would give you the string `"enum_variant"`, with the quotes, since a bare `enum_variant` token isn't valid JSON. But let me confirm real quick..

Reply:
Yep it will, but serde_json::Value::as_str() would unwrap that. Or `let Value::String(my_enum) = serde_json::to_value(&my_enum).unwrap() else { unreachable!() };`.

Reply:
Ahhhhhh, okay

```
>> serde_json::to_string(&InvoiceType::Usage).unwrap().as_str()
"\"usage\""
>> serde_json::to_value(InvoiceType::Usage).unwrap().as_str().unwrap()
"usage"
```

So `serde_json::Value::as_str()` gives the bare string. Neat!